examples/code/role_playing.py

```diff
@@ -0,0 +1,79 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from colorama import Fore
+
+from camel.societies import RolePlaying
+from camel.typing import TaskType
+from camel.utils import print_text_animated
+
+
+def main(model_type=None) -> None:
+    task_prompt = "Develop a poll app"
+    language = "JavaScript"
+    domain = "Sociology"
+    meta_dict = {"language": language, "domain": domain}
+    role_play_session = RolePlaying(
+        f"{language} Programmer",
+        f"Person working in {domain}",
+        task_prompt=task_prompt,
+        with_task_specify=True,
+        model_type=model_type,
+        task_type=TaskType.CODE,
+        extend_sys_msg_meta_dicts=[meta_dict, meta_dict],
```

Comment:
This is too complicated as an API. Very unintuitive. We need to segregate prompt engineering of AI society and Code from the dialogue algorithm of RolePlaying. I would create an instance of RolePair into the constructor of RolePlaying. AISocietyRoles and CodeRoles are the concrete classes inherited from RolePair. TDYT @lightaime

Reply:
This is the right place to apply **dependency inversion** to disentangle {AISocietyRoles and CodeRoles} from RolePlaying via the interface RolePair.